src/models/collections.rs

```diff
@@ -0,0 +1,109 @@
+use super::{
+    ids::{Base62Id, ProjectId},
+    users::UserId,
+};
+use crate::database;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+/// The ID of a specific collection, encoded as base62 for usage in the API
+#[derive(Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(from = "Base62Id")]
+#[serde(into = "Base62Id")]
+pub struct CollectionId(pub u64);
+
+/// A collection returned from the API
+#[derive(Serialize, Deserialize, Clone)]
+pub struct Collection {
+    /// The ID of the collection, encoded as a base62 string.
+    pub id: CollectionId,
+    /// The person that has ownership of this collection.
+    pub user: UserId,
+    /// The title or name of the collection.
+    pub title: String,
+    /// A short description of the collection.
+    pub description: String,
+
+    /// An icon URL for the collection.
+    pub icon_url: Option<String>,
+    /// Color of the collection.
+    pub color: Option<u32>,
+
+    /// The status of the collectin (eg: whether collection is public or not)
+    pub status: CollectionStatus,
+
+    /// The date at which the collection was first published.
+    pub created: DateTime<Utc>,
+
+    /// The date at which the collection was updated.
+    pub updated: DateTime<Utc>,
+
+    /// A list of ProjectIds that are in this collection.
+    pub projects: Vec<ProjectId>,
+}
+
+impl From<database::models::Collection> for Collection {
+    fn from(c: database::models::Collection) -> Self {
+        Self {
+            id: c.id.into(),
+            user: c.user_id.into(),
+            created: c.created,
+            title: c.title,
+            description: c.description,
+            updated: c.updated,
+            projects: c.projects.into_iter().map(|x| x.into()).collect(),
+            icon_url: c.icon_url,
+            color: c.color,
+            status: c.status,
+        }
+    }
+}
+
+/// A status decides the visibility of a collection in search, URLs, and the whole site itself.
+/// Listed - collection is displayed on search, and accessible by URL (for if/when search is implemented for collections)
+/// Unlisted - collection is not displayed on search, but accessible by URL
+/// Private - collection is approved, but is not viewable to the public
+#[derive(Serialize, Deserialize, Copy, Clone, Eq, PartialEq, Debug)]
+#[serde(rename_all = "lowercase")]
+pub enum CollectionStatus {
+    Listed,
+    Unlisted,
+    Private,
+    Unknown,
```

Comment:
I think we only need listed/unlisted for now and add a rejected status (for collections we take down).

Reply:
I can see uses for a private collection